### DISCUSSION TITLE
Remove Pi-Zero W instructions

### DISCUSSION THREAD

**Discussion by tkhan456** [mKeRix/room-assistant#335] [General]
It seems that Pi Zeros really can't run RA well.  From my personal experience and what appears to be showing up on many "Bug" reports and on posts in Reddit, I don't believe Pi Zeros can be utilized well for RA.  I switched my setup to Pi 3Bs and works wonderfully (so far).  Just a suggestion so others dont spend $$ on Pi Zeros only to have to either return them or waste $$

**Comment by mKeRix** [comment]
I think a lot of issues mention the Pi Zero solely because it's the main install base for room-assistant, not because it causes an unusually high amount of issues. The Pi 3B and the Pi Zero W have the same shared Bluetooth/WiFi chip if I remember correctly, which is the most important part. The extra processing power of the 3B helps with installing and loading some stuff faster, but shouldn't make a huge impact in the long run. I myself run a cluster of Pi Zero Ws successfully since a long time and I also know of a couple other users that have medium to large installations like that. 

I think removing the instructions would a) remove the choice from the user and b) drastically limit the appeal of the project for many. The Pi Zero costs a fraction of a full sized Raspberry, which adds up quickly if you plan on outfitting a whole house with them.

**Reply by shmookles** [reply]
I'm having issues with Pi Zero W where it will crash and no longer report tracking data via MQTT to HA. Is anyone seeing this behavior?

**Reply by mKeRix** [reply]
Can you maybe post some logs or the error that room-assistant crashes with?

**Comment by fidonet** [comment]
Hi

Not that I know of. 

I have 4 Pi Zero A’s dotted around my house and they are all reporting and all have fairly high up times. 

When you say “crash” do you mean the Modena process stops running or something else?

How are you starting the process? (From systemd or manually?)

If from systemd then I’d investigate why the watchdog function isn’t working for you

Mine often crash due to problems binding to wlan0 due to it being a shared interface (bug allegedly fixed in the latest release) but systemd catches that and restarts it fine each time

– 
Jon Morby

**Comment by shmookles** [comment]
so it just happened again where pi zero w hung, I cannot even SSH into. I can power cycle it and it will work, but I want to wait to make sure I grab the correct logs you all need. Can you please share with me what commands I need to run to grab the relevant logs for you all?

**Reply by mKeRix** [reply]
I think the most interesting logs are:

- room-assistant logs (run `journalctl -u room-assistant.service` and type shift + g to scroll to the bottom)
- kernel messages (run `dmesg`)
- content of `/var/log/syslog` and `/var/log/messages` towards the bottom (open with less and use shift + g to scroll down)

**Comment by shmookles** [comment]
Just happened again where I could not SSH into the PI and had to do a hard reboot by unplugging and plugging back in. After reboot I captured the logs you requested. Any chance you can have a look and let me know what might be the issue?

- journalctl -u room-assistant.service --> Link: https://pastebin.pl/view/9045bc7a

- dmesg --> Link: https://pastebin.pl/view/25fb9b31

- /var/log/syslog --> Link: https://pastebin.pl/view/ed99be4a

- /var/log/messages --> Link: https://pastebin.pl/view/53e58c69

**Reply by mKeRix** [reply]
Unfortunately these logs are not persisted across reboots on Raspbian. From what you posted I would at least try playing with the settings a bit to see if that resolves your issue... try raising the `interval` to 10 and the `scanTimeLimit` to 6 for the Bluetooth Classic integration. These values are not as aggressive on the shared Bluetooth/WiFi chip, so maybe that already does the trick.
